plasma_framework/migrations/5_deploy_and_register_payment_exit_game.js

```diff
@@ -0,0 +1,90 @@
+const Erc20Vault = artifacts.require('Erc20Vault');
+const OutputGuardHandlerRegistry = artifacts.require('OutputGuardHandlerRegistry');
+const SpendingConditionRegistry = artifacts.require('SpendingConditionRegistry');
+const PaymentExitGame = artifacts.require('PaymentExitGame');
+const PaymentChallengeStandardExit = artifacts.require('PaymentChallengeStandardExit');
+const PaymentChallengeIFENotCanonical = artifacts.require('PaymentChallengeIFENotCanonical');
+const PaymentChallengeIFEInputSpent = artifacts.require('PaymentChallengeIFEInputSpent');
+const PaymentChallengeIFEOutputSpent = artifacts.require('PaymentChallengeIFEOutputSpent');
+const PaymentOutputGuardHandler = artifacts.require('PaymentOutputGuardHandler');
+const PaymentOutputToPaymentTxCondition = artifacts.require('PaymentOutputToPaymentTxCondition');
+const PaymentStartInFlightExit = artifacts.require('PaymentStartInFlightExit');
+const PaymentStartStandardExit = artifacts.require('PaymentStartStandardExit');
+const PaymentTransactionStateTransitionVerifier = artifacts.require('PaymentTransactionStateTransitionVerifier');
+const PaymentPiggybackInFlightExit = artifacts.require('PaymentPiggybackInFlightExit');
+const PaymentProcessStandardExit = artifacts.require('PaymentProcessStandardExit');
+const PaymentProcessInFlightExit = artifacts.require('PaymentProcessInFlightExit');
+const PlasmaFramework = artifacts.require('PlasmaFramework');
+
+module.exports = async (deployer) => {
+    const PAYMENT_TX_TYPE = 1;
+    const PAYMENT_OUTPUT_TYPE = 1;
+    const MORE_VP_PROTOCOL = 2;
+
+    await deployer.deploy(PaymentChallengeStandardExit);
+    await deployer.deploy(PaymentChallengeIFENotCanonical);
+    await deployer.deploy(PaymentChallengeIFEInputSpent);
+    await deployer.deploy(PaymentChallengeIFEOutputSpent);
+    await deployer.deploy(PaymentStartInFlightExit);
+    await deployer.deploy(PaymentStartStandardExit);
+    await deployer.deploy(PaymentPiggybackInFlightExit);
+    await deployer.deploy(PaymentProcessStandardExit);
+    await deployer.deploy(PaymentProcessInFlightExit);
+
+    await deployer.link(PaymentChallengeStandardExit, PaymentExitGame);
+    await deployer.link(PaymentChallengeIFENotCanonical, PaymentExitGame);
+    await deployer.link(PaymentChallengeIFEInputSpent, PaymentExitGame);
+    await deployer.link(PaymentChallengeIFEOutputSpent, PaymentExitGame);
+    await deployer.link(PaymentStartInFlightExit, PaymentExitGame);
+    await deployer.link(PaymentStartStandardExit, PaymentExitGame);
+    await deployer.link(PaymentPiggybackInFlightExit, PaymentExitGame);
+    await deployer.link(PaymentProcessStandardExit, PaymentExitGame);
+    await deployer.link(PaymentProcessInFlightExit, PaymentExitGame);
+
+    await deployer.deploy(OutputGuardHandlerRegistry);
+    await deployer.deploy(SpendingConditionRegistry);
+    await deployer.deploy(PaymentTransactionStateTransitionVerifier);
+
+    await deployer.deploy(
```

Comment:
I'm not too familiar with the Plasma logic, could you elaborate on why `PaymentExitGame` is a confusing name for future development?

Reply:
oh, the naming of `PaymentExitGame` is fine. The issue is that the plan of ALD is to reuse the same contract code multiple times. As a result, we would deploy `PaymentExitGame` contract, for instance, with another `PAYMENT_V2_TX_TYPE`.

However, truffle seems like relying on the contract name itself to know which contract its indicating, for instance  we do `PaymentExitGame.address` this when we need the address. So the issue is that there would be multiple contract instance of `PaymentExitGame` and we would need to find a way to solve this.

Reply:
Okay, I gotcha, might need to look into a factory system if you want to keep this clean

Reply:
I update some commits trying to solve this problem. Aside from the `PlasmaFramework`, all other contracts uses `new` instead to solve this. They would be registered to the `PlasmaFramework` anyway so would be able to get the address out.

However, this approach has clear downside that it would not show the gas cost in summary.